datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/dataproduct/DataProductEntitiesResolver.java

```diff
@@ -0,0 +1,94 @@
+package com.linkedin.datahub.graphql.resolvers.dataproduct;
+
+import com.google.common.collect.ImmutableList;
+import com.linkedin.datahub.graphql.QueryContext;
+import com.linkedin.datahub.graphql.generated.DataProduct;
+import com.linkedin.datahub.graphql.generated.DataProductEntitiesInput;
+import com.linkedin.datahub.graphql.generated.SearchResults;
+import com.linkedin.datahub.graphql.resolvers.EntityTypeMapper;
+import com.linkedin.datahub.graphql.types.mappers.UrnSearchResultsMapper;
+import com.linkedin.entity.client.EntityClient;
+import com.linkedin.metadata.query.filter.Condition;
+import com.linkedin.metadata.query.filter.ConjunctiveCriterion;
+import com.linkedin.metadata.query.filter.ConjunctiveCriterionArray;
+import com.linkedin.metadata.query.filter.Criterion;
+import com.linkedin.metadata.query.filter.CriterionArray;
+import com.linkedin.metadata.query.filter.Filter;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static com.linkedin.datahub.graphql.resolvers.ResolverUtils.*;
+import static com.linkedin.datahub.graphql.resolvers.search.SearchUtils.SEARCHABLE_ENTITY_TYPES;
+
+
+/**
+ * Retrieves a list of entities associated with a given data product
+ */
+@Slf4j
+public class DataProductEntitiesResolver implements DataFetcher<CompletableFuture<SearchResults>> {
+
+  private static final String DATA_PRODUCT_FIELD_NAME = "dataProducts";
+  private static final String INPUT_ARG_NAME = "input";
+  private static final String DEFAULT_QUERY = "*";
+  private static final Integer DEFAULT_START = 0;
+  private static final Integer DEFAULT_COUNT = 20;
+  private static final DataProductEntitiesInput DEFAULT_ENTITIES_INPUT = new DataProductEntitiesInput();
+
+  static {
+    DEFAULT_ENTITIES_INPUT.setQuery(DEFAULT_QUERY);
+    DEFAULT_ENTITIES_INPUT.setStart(DEFAULT_START);
+    DEFAULT_ENTITIES_INPUT.setCount(DEFAULT_COUNT);
+  }
+
+  private final EntityClient _entityClient;
+
+  public DataProductEntitiesResolver(final EntityClient entityClient) {
```

Comment:
Do we actually use this field on the frontend? I know for domains we ended up using search mainly

Reply:
oh woops this is a remnant before some changes happened! will remove